pkg/discovery/network/generic.go

```diff
@@ -53,13 +55,64 @@ func findClusterIPRange(clientSet kubernetes.Interface) (string, error) {
 		return clusterIPRange, nil
 	}
 
+	clusterIPRange, err = findClusterIPRangeSvcCreation(clientSet)
+	if err == nil && clusterIPRange != "" {
+		return clusterIPRange, nil
+	}
+
 	return "", fmt.Errorf("clusterIPRange can't be found by any methods")
 }
 
 func findClusterIPRangeApiserver(clientSet kubernetes.Interface) (string, error) {
 	return findPodCommandParameter(clientSet, "component=kube-apiserver", "--service-cluster-ip-range")
 }
 
+func findClusterIPRangeSvcCreation(clientSet kubernetes.Interface) (string, error) {
+	// find service cidr based on https://stackoverflow.com/questions/44190607/how-do-you-find-the-cluster-service-cidr-of-a-kubernetes-cluster
+	invalidSvcSpec := &v1.Service{
+		ObjectMeta: v1meta.ObjectMeta{
+			Name: "invalid-svc",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "1.1.1.1",
+			Ports: []v1.ServicePort{
+				{
+					Port: 443,
+					TargetPort: intstr.IntOrString{
+						IntVal: 443,
+					},
+				},
+			},
+		},
+	}
+	_, err := clientSet.CoreV1().Services("submariner-operator").Create(invalidSvcSpec)
```

Comment:
Creating svc in `submariner-operator` namespace might fail if it is called before the namespace is created. It seems to happen even on `subctl join` command. Other candidate will be `kube-system`, `default`, and random namespace created just for this check. Any ideas?

Reply:
I think we could use empty namespace but, if not, `default`.

Reply:
This code is called from both command line and operator.
subctl actually calls it on subctl join before submariner-namespace is created, so it shouldn't create service in submariner-operator namespace. On the other hand, the operator is only allowed to create resources inside submariner-operator namespace, so it shouldn't create service in default namespace.

Therefore, the code was changed to check whether it is running in operator and to create service in the proper namespace.